bench-vortex/src/bin/clickbench.rs

```diff
@@ -23,9 +24,18 @@ use vortex_datafusion::persistent::metrics::VortexMetricsFinder;
 
 feature_flagged_allocator!();
 
+#[derive(ValueEnum, Clone, Copy, Debug, Hash, Default, PartialEq, Eq)]
+enum Engine {
+    #[default]
+    DataFusion,
+    DuckDB,
+}
+
 #[derive(Parser, Debug)]
 #[command(version, about, long_about = None)]
 struct Args {
+    #[arg(long, value_enum, default_value_t = Engine::DataFusion)]
+    engine: Engine,
```

Comment:
there are some flags in the runner that would only work with datafusion. It might be worth at least to log if duckdb is selected as the engine and things like threads or emulate_object_store are set

Reply:
honestly might be time to remove `emulate_object_store`, IDK if anyone actually uses it. 
duckdb does seem to [have](https://duckdb.org/docs/stable/configuration/overview.html#examples) a way to control the number of threads, IDK how hard it is to wire that in

Reply:
Yeah that's a good idea flagging if the combinations are invalid. For this PR I think it's fine to keep as is and revisit this once DuckDB is fully wired in. E.g. threads might also be used by DuckDB etc.

Reply:
> honestly might be time to remove `emulate_object_store`, IDK if anyone actually uses it. duckdb does seem to [have](https://duckdb.org/docs/stable/configuration/overview.html#examples) a way to control the number of threads, IDK how hard it is to wire that in

Just saw this reply. I'm not sure yet what the implications are about removing that flag tbh. But happy to clean up more flags eventually.